File: retos/Reto-2/Darik15ing/palindromo.js

function palindromo (text){
    if (text.length > 1 && typeof text){
        //coembjgdfjzhf
        let textOriginal=text;
        text=text.normalize("NFD").replace(/[\u0300-\u036f]/g, "");
        text=text.replaceAll(/\s/g,"");
        text=text.replaceAll(/\W/g,"");

        let textInverso = text.split("").reverse();
        textInverso = textInverso.join();
        textInverso = textInverso.replaceAll(",","");

        let palabra=""
        if (textOriginal.search(" ") !== -1){
            palabra = "La frase ";
        }
        else{
            palabra = "La palabra ";
        }
        
        if (text.toLocaleLowerCase() === textInverso.toLocaleLowerCase()){
            return palabra + "es palíndromo";
        }
        else{
            return palabra + "no es palíndromo";
        }
    }
    return "No es posible formar un palíndromo con un caracter o números"
}    
 
module.exports = palindromo;

File: retos/Reto-2/NitroXen/palindromo.py
import re
from unicodedata import normalize


def palindrome(word):
    word = word.upper()
    if isAPhrase(word):
        word = word.replace(" ", "").strip("¿").strip("?").strip(
            "¡").strip("!").strip("'").strip('"').strip(",").strip(".")
        word = re.sub(
            r"([^n\u0300-\u036f]|n(?!\u0303(?![\u0300-\u036f])))[\u0300-\u036f]+", r"\1",
            normalize("NFD", word), 0, re.I
        )
        word = normalize("NFC", word)
        if word == word[::-1]:
            return "la frase es un palindromo"
        else:
            return "la frase NO es un palindromo"

    else:
        if word == word[::-1]:
            return "la palabra es un palindromo"
        else:
            return "la palabra NO es un palindromo"


def isAPhrase(word):
    return " " in word
